vehicles/config/generate_all.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+# ---------------------------------------------------------------------- #
+#                                                                        #
+#  Copyright (C) 2022                                                    #
+#  Illini RoboMaster @ University of Illinois at Urbana-Champaign.       #
+#                                                                        #
+#  This program is free software: you can redistribute it and/or modify  #
+#  it under the terms of the GNU General Public License as published by  #
+#  the Free Software Foundation, either version 3 of the License, or     #
+#  (at your option) any later version.                                   #
+#                                                                        #
+#  This program is distributed in the hope that it will be useful,       #
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+#  GNU General Public License for more details.                          #
+#                                                                        #
+#  You should have received a copy of the GNU General Public License     #
+#  along with this program. If not, see <http://www.gnu.org/licenses/>.  #
+#                                                                        #
+# ---------------------------------------------------------------------- #
+
+'''
+Generate .h file from YAML files
+NOTE: for int, this convertor generate const float
+      for str, this convertor generate const int16_t based on dictionary.yaml
+Usage: ./generate_all.py index
+       where index is a yaml list
+'''
+
+# USER PARAMETERS START //
+
+# output dir for the generated .h file
+# if empty, use the same dir as the given YAML file
+output_dir = "./"
+
+# dictionary dir, convert str in YAML to int in .H
+dict_dir = "./dictionary.yaml"
```

Comment:
Is it possible that these can also be passed in as command line arguments?

Reply:
There is no need. `output_dir` and the `dict_dir` never change as soon as we decide them.